### ISSUE TITLE
[jitlib] ProxySpace should .stop proxies before removing them

### ISSUE THREAD

**Issue by jamshark70** [supercollider/supercollider#4394]
## Environment

* SuperCollider version: 3.10.2
* Operating system: Ubuntu Studio 18.04

## Steps to reproduce

```supercollider
p = ProxySpace.new.push;

~src = { |ffreq = 2000|
	(LPF.ar(LFSaw.ar(100), ffreq) * 0.1).dup
};
~src.play;

~ffreqMod = { LFDNoise3.kr(0.2).exprange(500, 1800) };
~src.set(\ffreq, ~ffreqMod);

p.pop;
p.clear;  // "pop" sound
```

## Expected vs. actual behavior

Actual: The filter makes a loud "pop" when clearing the proxy space.

This is because proxies within the space are cleared in an arbitrary order, without considering dependencies. I found locally in a couple of cases that the filter frequency proxy `~ffreqMod` was cleared first, setting the filter frequency to 0, which of course is a no-no for filter math.

Expected: The proxy space should go silent, transparently.

Tracking the dependencies will be difficult/impossible. But, an easy solution would be to call `.stop` on all proxies before clearing them. The filter will still glitch, but it won't be connected to the speakers.

**Comment by jamshark70** [comment]
Also, I recommend making one pass to stop proxies, and another to clear them, because a proxy might be only indirectly audible (e.g. filter feeds into distortion into chorus and only the chorus is played). Just adding `.stop.clear` into the existing loop would not help that case -- they all should be stopped before removing any of them.

**Comment by telephon** [comment]
>Tracking the dependencies will be difficult/impossible. 

You can track the dependencies (which may not be enough if you have loops of course). This is done in `getFamily`, which however returns an unordered collection. 

>But, an easy solution would be to call .stop on all proxies before clearing them. The filter will still glitch, but it won't be connected to the speakers.

The easiest is to get all those that are playing by `proxyspace.monitors`, and first clear them.

I'm not sure which of the two options is better: we could add a method that gives you the dependency tree of any proxy, and then traverse this. Or we could go with the simple one that just clears the monitors.

**Comment by jamshark70** [comment]
I guess objectively speaking, the best is to prevent the filter from glitching in the first place (by following dependencies). But complexity may also become a maintenance problem.

For awhile, I've been using JITLib like a modular synthesizer. This means more dependencies between proxies. Likely this was not the original use case for JITLib -- useful in any case to expose some limitations.

**Comment by telephon** [comment]
No, the idea has always been to have lots of small units that interconnect. And these glitches have been a problem every once in a while, so better we fix them properly.

It may not be a maintenance problem, it is mainly about adding a method and a unit test.

**Comment by jamshark70** [comment]
I'm happy to look at that, but it'll have to be a few days later.

I'll have other improvements to propose based on https://github.com/jamshark70/JITModular -- after hacking some internals, I've got it working to save and reload an entire ProxySpace, with quant, nodemaps, Halo specs, MIDI controller mappings (not inherently part of ProxySpace) and buffer references. I think it's an interesting middle layer between pure code and save-able patches (but I had to hack some fixes to get there, currently in a rather naughty way but would be worth pulling those into JITLib proper, when I have time to make PRs). It is really nice to be able to `p = JITModPatch.load`, pick the file, and everything is where you left it.

**Comment by telephon** [comment]
great! Whatever doesn't fit into the core usually goes into `JITLibExtensions`.
